applications/DEMApplication/python_scripts/sphere_strategy.py

```diff
@@ -319,12 +320,41 @@ def SolveSolutionStep(self):
         self.FixExternalForcesManually(time)
         (self.cplusplus_strategy).Solve()
 
-    def AdvanceInTime(self, step, time):
+    def AdvanceInTime(self, step, time, is_time_to_print = False):
         step += 1
-        time = time + self.dt
-
+        time += self.dt
+        self.UpdateTimeInModelParts(time, step, is_time_to_print)
         return step, time
 
+    def MoveAllMeshes(self, time, dt):
```

Comment:
also this is protected (or private)

if protected it should start with _
if private it should start with __